Source/Core/Content/ContentSystem.cpp

```diff
@@ -682,6 +678,47 @@ bool ContentSystem::RenameContentItemFile(ContentItem* contentItem, StringParam
   return true;
 }
 
+bool ContentSystem::MoveContentItem(ContentItem* contentItem, ContentLibrary* targetLibrary)
+{
+    // Build Strings
+
+    // Get the original full path to the content item and meta file
+    String oldId = contentItem->UniqueFileId;
+    String contentFile = contentItem->GetFullPath();
+    String contentMetaFile = contentItem->GetMetaFilePath();
+    
+    String oldPath = FilePath::GetDirectoryPath(contentFile);
+    String newFullPath = FilePath::Combine(targetLibrary->SourcePath, contentItem->Filename);
+
+    ContentLibrary* currentLibrary = contentItem->mLibrary;
+
+    // Does a file with this name already exist?
+    // Does a file with the same name exist in the target library folder? If not, then we need to move it.
+
+    bool fileExistsAtNewDirectory = FileExists(newFullPath);
+    if (fileExistsAtNewDirectory && (currentLibrary->SourcePath != targetLibrary->SourcePath))
+    {
+        return false;
+    }
+    else if (!fileExistsAtNewDirectory)
+    {
+        // If the file does not exists in the directory that we are trying to move it to, we need to remove the meta file and move the file.
+        if (!contentItem->DeleteMetaFile())
+            PlasmaPrint("Unable to delete meta file for %s", contentItem->Filename);
+        MoveFile(newFullPath, contentItem->GetFullPath());
+    }
+
+    currentLibrary->ContentItems.Erase(oldId);
+    contentItem->UniqueFileId = UniqueFileId(newFullPath);
+    contentItem->mLibrary = targetLibrary;
+    targetLibrary->ContentItems.Insert(contentItem->UniqueFileId, contentItem);
+
+    currentLibrary->Load();
+    targetLibrary->Load();
+
+    return true;
```

Comment:
A slight worry I have is that if this is called outside of `ContentSystem::MoveContentItem` then `contentItem->SaveContent()` won't get executed. Perhaps this function should be made protected instead of public in `ContentSystem`

Reply:
SaveContent definitely should be executed here. The UpdateContentItem function thats run back in the resourceOperations.cpp MoveResource call doesn't modify anything within contentItem like I thought it did. Good catch.